episodes/4-advanced-layer-types.Rmd

```diff
@@ -785,6 +787,220 @@ This is called hyperparameter tuning.
 ::::
 :::
 
+## Hyperparameter tuning
+::: instructor
+## Do a live demo instead of live coding
+You might want to demonstrate this section on hyperparamater tuning instead of doing live coding.
+The goal is to show that hyperparameter tuning can be done easily with `keras_tuner`, not to memorize all the exact syntax of how to do it. This will probably save you half an hour of participants typing over code that they already know from before. In addition, on really slow machines running the grid search could possibly take more than 10 minutes.
+:::
+
+In general if you are varying hyperparameters (such as the dropout rate) to find the combination of hyperparameters with the best model performance this is called hyperparameter tuning. A naive way to do this is to write a for-loop and train a slightly different model in every cycle. 
+However, it is better to use the `keras_tuner` package for this.
+
+Let's first define a function that creates a neuronal network given 2 hyperparameters, namely the dropout rate and the number of layers:
+```python
+def create_nn_with_hp(dropout_rate, n_layers):
+    inputs = keras.Input(shape=train_images.shape[1:])
+    x = inputs
+    for layer in range(n_layers):
+        x = keras.layers.Conv2D(50, (3, 3), activation='relu')(x)
+        x = keras.layers.MaxPooling2D((2, 2))(x)
+    x = keras.layers.Dropout(dropout_rate)(x)
+    x = keras.layers.Flatten()(x)
+    x = keras.layers.Dense(50, activation='relu')(x)
+    outputs = keras.layers.Dense(10)(x)
+    model = keras.Model(inputs=inputs, outputs=outputs, name="cifar_model")
```

Comment:
Just a reminder for later: this (and all other references to `cifar_model`) needs to be adapted to the changes in #445 .

Reply:
Yes, this is done in #448